pkg/controllers/user/monitoring/clusterHandler.go

```diff
@@ -419,7 +420,17 @@ func (ch *clusterHandler) deployApp(appName, appTargetNamespace string, appProje
 		},
 	}
 
-	_, err = utils.DeployApp(ch.app.cattleAppClient, appDeployProjectID, app, false)
+	// redeploy cluster-monitoring App forcibly if its workload doesn't exist
+	var forceRedeploy bool
+	appWorkload, err := ch.app.agentStatefulSetClient.GetNamespaced(appTargetNamespace, fmt.Sprintf("prometheus-%s", appName), metav1.GetOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		return nil, errors.Wrapf(err, "failed to get statefulset %s/prometheus-%s", appTargetNamespace, appName)
+	}
+	if appWorkload == nil || appWorkload.Name == "" || appWorkload.DeletionTimestamp != nil {
```

Comment:
Also checking apierrors.IsNotFound(err) as well would be decisive in most cases?

Reply:
Line 426 already checks for when error is not NotFound and returns it

Reply:
Yes that is the assumption that appWorkload will be nil; but I have seen with client-go for ingress object that along with NotFound error the object returned is non-nil at times. Hence feel that also checking for the "error is NotFound" would be reliable. If your tests show that object returned is always nil, that is good.

Reply:
Did you see a nil object always? @mrajashree

Reply:
@prachidamle I was going by [this](https://github.com/rancher/types/blob/master/apis/apps/v1/zz_generated_stateful_set_controller.go#L116-L134) and it always returns nil if there's an error. 
But I also checked in my setup and the object is nil for NotFound error.

I think listers return nil with error, maybe not the client's [GetNamespaced](https://github.com/rancher/rancher/blob/master/vendor/github.com/rancher/norman/objectclient/object_client.go#L142-L158) call

Reply:
great, thanks for digging it up